tests/scenario/test_transition_to_https.py

```diff
@@ -0,0 +1,129 @@
+# Feature: Blog
+#     A site where you can publish your articles.
+#
+#     Scenario: Publishing the article
+#         Given I'm an author user
+#         And I have an article
+#
+#         When I go to the article page
+#         And I press the publish button
+#
+#         Then I should not see the error message
+#         And the article should be published  # Note: will query the database
+
+
+# https://stackoverflow.com/a/62176555/3516684
+
+
+# | Gherkin element | pytest artifact |
+# |-----------------|-----------------|
+# | Feature         | *.py            |
+# | Scenario        | Class           |
+# | Step            | Method          |
+
+
+"""Feature: The workload's scheme is reflected in the pebble command and in relation data.
+
+This feature spans:
+- manifest generation (pebble layer)
+- schema generation (alertmanager_dispatch provider)
+
+The alertmanager server can serve over HTTP or HTTPS. The requirer side of the relation may be
+design to take URL parts rather than a full URL. Prometheus takes URL parts and would need to
+generate its "alertmanagers" config section differently depending on the scheme.
+"""
+
+import json
+from unittest.mock import patch
+
+import pytest
+from helpers import begin_with_initial_hooks_isolated, add_relation_sequence
+from scenario import Relation, State, Context
+
+
+@pytest.fixture(params=["localhost", "am-0.endpoints.cluster.local"])
+def fqdn(request):
+    yield request.param
+
+
+@pytest.fixture(autouse=True)
+def patch_fqdn(fqdn):
+    with patch("socket.getfqdn", new=lambda *args: fqdn):
+        yield
+
+
+@pytest.fixture(params=[True, False])
+def is_leader(request):
+    yield request.param
+
+
+@pytest.fixture
+def initial_state(context, is_leader):
+    """An isolated alertmanager instance after the startup sequence."""
+    return begin_with_initial_hooks_isolated(context, leader=is_leader)
+
+
+@pytest.fixture
+def state_with_alerting_relation(initial_state, context, is_leader):
+    """Alertmanager after an alerting relation has been added."""
+    return add_relation_sequence(context, initial_state, Relation("alerting", relation_id=10))
+
```

Comment:
With all these fixtures in place it seems a bit too deconstructed.
I find it hard to reason about the tests with such a deconstruction.